File: src/main/java/brasil/entity/Zone.java
package brasil.entity;

import java.util.ArrayList;

public class Zone {
    private String nom;
    private double prixLivraison;
    private int id;   
    private ArrayList<Quartier> quartiers;
    public Zone(String nom, double prixLivraison) {
        this.nom = nom;
        this.prixLivraison = prixLivraison;
    }
    public String getNom() {
        return nom;
    }
    public void setNom(String nom) {
        this.nom = nom;
    }
    public double getPrixLivraison() {
        return prixLivraison;
    }
    public void setPrixLivraison(double prixLivraison) {
        this.prixLivraison = prixLivraison;
    }
    public int getId() {
        return id;
    }
    public void setId(int id) {
        this.id = id;
    }
    public ArrayList<Quartier> getAllQuartiers() {
        return quartiers;
    }
    public void addQuartier(Quartier quartier) {
        this.quartiers.add(quartier);
    }
    
}

File: src/Entity/Zone.php
<?php

namespace App\Entity;

use App\Repository\ZoneRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ZoneRepository::class)]
class Zone
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 100)]
    private ?string $nom ;

    #[ORM\Column(name: "prix_livraison", type: Types::DECIMAL, precision: 10, scale: 2)]
    private ?string $prixLivraison = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): static
    {
        $this->nom = $nom;

        return $this;
    }

    public function getPrixLivraison(): ?string
    {
        return $this->prixLivraison;
    }

 
}
